superset-frontend/src/components/Select/Select.tsx

```diff
@@ -253,25 +262,30 @@ const Select = forwardRef(
         });
       }
       setInputValue('');
+      fireOnChange();
+      onSelect?.(selectedItem, option);
     };
 
     const clear = () => {
-      setSelectValue(
-        fullSelectOptions
-          .filter(
-            option => option.disabled && hasOption(option.value, selectValue),
-          )
-          .map(option =>
-            labelInValue
-              ? { label: option.label, value: option.value }
-              : option.value,
-          ),
-      );
+      if (isSingleMode) {
```

Comment:
Single mode was previously returning an empty array but it should be `undefined`.